http/file_server.ts

```diff
@@ -516,6 +516,15 @@ export async function serveDir(req: Request, opts: ServeDirOptions = {}) {
           const path = posix.join(fsPath, "index.html");
           const indexFileInfo = await Deno.lstat(path);
           if (indexFileInfo.isFile) {
+            // If the current URL's pathname doesn't end with a slash, any
+            // relative URLs in the index file will resolve against the parent
+            // directory, rather than the current directory. To prevent that, we
+            // return a 301 redirect to the URL with a slash.
+            const url = new URL(req.url);
+            if (!url.pathname.endsWith("/")) {
```

Comment:
Can this check be performed with some existing variable? For example, does `fsPath.endsWith("/")` work?

Reply:
If so, we can `new URL` in the if block and keep the perf a bit better

Reply:
I thought `fsPath.endsWith("/")` would not work because of the path normalization, but I've checked it, and it doesn't seem to remove trailing slashes. However, creating a `URL` object is pretty much needed inside the `if` block, since doing `req.url + "/"` will not work if there's a query string.